web/src/routes/admin/library-management/+page.svelte

```diff
@@ -243,6 +243,19 @@
     updateLibraryIndex = selectedLibraryIndex;
   };
 
+  const onToggleReadOnlyClicked = async () => {
+    const library = libraries[selectedLibraryIndex];
+    const prevValue = library.isReadOnly == null ? library.type === LibraryType.External : library.isReadOnly;
+
+    try {
+      await updateLibrary({ id: library.id, updateLibraryDto: { isReadOnly: !prevValue } });
+      closeAll();
+      await readLibraryList();
+    } catch (error) {
+      handleError(error, 'Unable to update library');
+    }
+  };
+
```

Comment:
Just wondering, but how would this code behave if a user clicked multiple times in quick succession?

Reply:
it might be a bit wasteful to do, but nothing jumps out at me as being particularly problematic if that were to occur